lib/src/capture_screen.dart

```diff
@@ -74,68 +67,42 @@ Future<List<int>> takeScreenshot(
     String screenshotName,
     {Duration? timeout}) async {
   if (const bool.fromEnvironment('dart.library.js_util')) {
-    // kIsWeb implementation
-    throw 'web not yet implemented';
-    // await binding.takeScreenshot(name);
-    // return;
+    throw Exception('web not yet implemented');
   } else if (Platform.isAndroid) {
-    try {
-      // await integrationTestChannel.invokeMethod<void>(
-      //   'convertFlutterSurfaceToImage',
-      // );
-      await binding.convertFlutterSurfaceToImage();
-    } catch (e) {
-      print(
-          'converting surface to image failed; probably it was already an image?: $e');
-    }
-    final endTime = binding.clock
-        .fromNowBy(timeout ?? const Duration(seconds: 30)) as DateTime;
-    do {
-      if ((binding.clock.now() as DateTime).isAfter(endTime)) {
-        break;
-      }
-      await binding.pump(const Duration(milliseconds: 100));
-    } while (binding.hasScheduledFrame as bool);
+    await integrationTestChannel.invokeMethod<void>(
+      'convertFlutterSurfaceToImage',
+    );
   }
-  final reportData = <String, dynamic>{};
-  reportData['screenshots'] ??= <dynamic>[];
-  final data = await _takeScreenshot(
-      binding, integrationTestChannel, platformDispatcher, screenshotName);
-  assert(data.containsKey('bytes'));
-
-  (reportData['screenshots']! as List<dynamic>).add(data);
-  return data['bytes']! as List<int>;
-}
-
-Future<Map<String, dynamic>> _takeScreenshot(
-    dynamic binding,
-    dynamic integrationTestChannel,
-    dynamic platformDispatcher,
-    String screenshot) async {
-  integrationTestChannel.setMethodCallHandler(
-      (dynamic call) => _onMethodChannelCall(call, platformDispatcher));
-  // final rawBytes = await integrationTestChannel.invokeMethod<List<int>>(
-  //   'captureScreenshot',
-  //   <String, dynamic>{'name': screenshot},
-  // ) as List<int>?;
-  final rawBytes = await binding.takeScreenshot(screenshot);
-
+  await binding.pump();
+  binding.reportData ??= <String, dynamic>{};
+  binding.reportData!['screenshots'] ??= <dynamic>[];
+  integrationTestChannel.setMethodCallHandler((dynamic call) async {
```

Comment:
The original method used in master has two parameters.
This was causing an error, not sure if it is required by patrol, but this is the right signature with the `flutter_test` implementation.

Reply:
To be honest, I don't know what method exactly you mean, but I will just resolve this conversation.
I will run my test pipeline after this and we will see if anything breaks with patrol.